app/build.gradle

```diff
@@ -66,6 +68,33 @@ android {
     }
 }
 
+android.applicationVariants.all { variant ->
+    def buildType = variant.buildType.name
+
+    if (buildType == "release") {
+        def versionCode = AppVersion.baseVersionCode
+
+        // The Google Play Store does not allow multiple APKs for the same app that all have the
+        // same version code. Therefore we need to have different version codes for our ARM and x86
+        // builds.
+
+        // Our x86 builds need a higher version code to avoid installing ARM builds on an x86 device
+        // with ARM compatibility mode. In addition to that aarch64 builds should be preferred over
+        // ARM builds if suitable: x86 (+2) > aarch64 (+1) > arm.
+
+        // Note: This assumes that we do not need to release more often than every 2 minutes!
+        //       Otherwise this will create version code conflicts.
+
+        if (variant.flavorName.contains("X86")) {
+            versionCode = versionCode + 2
```

Comment:
Should we use a digit to highlight what architecture it is? I'm worried that `2531704` means nothing architecture-wise. This can be confusing on the Google Play dashboard. How about we have:

* `25317040`  -> arm
* `25317041` -> arm64
* `25317042` -> x86

What do you think?

Reply:
I implemented the proposal.